tests/tests_unit/test_api/test_assets.py

```diff
@@ -314,237 +310,175 @@ def test_assets_update_object(self):
-        assert 2 == len(mock_post_asset_hierarchy.calls)
-
-    @pytest.fixture
-    def mock_post_assets_failures(self, cognite_client, rsps):
-        def request_callback(request):
-            items = jsgz_load(request.body)["items"]
-            response_assets = []
-            item = items[0]
-            parent_id = None
-            if "parentId" in item:
-                parent_id = item["parentId"]
-            if "parentExternalId" in item:
-                parent_id = item["parentExternalId"] + "id"
-            id = item.get("refId", "root_") + "id"
-            response_assets.append(
-                {
-                    "id": id,
-                    "parentId": parent_id,
-                    "externalId": item["externalId"],
-                    "parentExternalId": item.get("parentExternalId"),
-                }
-            )
-
-            if item["name"] == "400":
-                return 400, {}, json.dumps({"error": {"message": "user error", "code": 400}})
-
-            if item["name"] == "500":
-                return 500, {}, json.dumps({"error": {"message": "internal server error", "code": 500}})
-
-            return 200, {}, json.dumps({"items": response_assets})
-
-        rsps.add_callback(
-            rsps.POST,
-            cognite_client.assets._get_base_url_with_base_path() + "/assets",
-            callback=request_callback,
-            content_type="application/json",
-        )
-        with set_request_limit(cognite_client.assets, 1):
-            yield rsps
-
-    def test_post_with_failures(self, cognite_client, mock_post_assets_failures):
-        assets = [
-            Asset(name="200", external_id="0"),
-            Asset(name="200", parent_external_id="0", external_id="01"),
-            Asset(name="400", parent_external_id="0", external_id="02"),
-            Asset(name="200", parent_external_id="02", external_id="021"),
-            Asset(name="200", parent_external_id="021", external_id="0211"),
-            Asset(name="500", parent_external_id="0", external_id="03"),
-            Asset(name="200", parent_external_id="03", external_id="031"),
-        ]
-        with pytest.raises(CogniteAPIError) as e:
-            cognite_client.assets.create_hierarchy(assets)
-
-        assert {a.external_id for a in e.value.unknown} == {"03"}
-        assert {a.external_id for a in e.value.failed} == {"02", "021", "0211", "031"}
-        assert {a.external_id for a in e.value.successful} == {"0", "01"}
-
-    def test_post_with_all_failures_and_more_requests_than_workers(self, cognite_client, mock_post_assets_failures):
-        num_assets = cognite_client.config.max_workers * 10
-        assets = [Asset(name="400", external_id=str(i)) for i in range(num_assets)]
-        with pytest.raises(CogniteAPIError) as e:
-            cognite_client.assets.create_hierarchy(assets)
-
-        assert {a.external_id for a in e.value.failed} == {str(i) for i in range(num_assets)}
+# class TestAssetPoster:
```

Comment:
I think instead of revamping these unit tests, we should instead extend the integration tests for creating hierarchies. Will make it easier to evolve the implementation.